ack/utils/image_utils.py

```diff
@@ -0,0 +1,393 @@
+    current_pixel_sizes = np.array(current_pixel_sizes)
+    desired_pixel_sizes = np.array(desired_pixel_sizes)
+
+    # Only resize raw image if desired pixel sizes is different from current
+    if not np.array_equal(current_pixel_sizes, desired_pixel_sizes):
+        scale_raw = current_pixel_sizes / desired_pixel_sizes
+        raw = np.stack([proc.resize(channel, scale_raw, "bilinear") for channel in raw])
+
+    # Prep segmentations
+    nuc_seg = nuc_seg.get_image_data("YXZ", S=0, T=0, C=0)
+    memb_seg = memb_seg.get_image_data("YXZ", S=0, T=0, C=0)
+
+    # We do not assume that the segementations are the same size as the raw
+    # Resize the segmentations to match the raw
+    # We drop the channel dimension from the raw size retrieval
+    raw_size = np.array(raw.shape[1:]).astype(float)
+    nuc_size = np.array(nuc_seg.shape).astype(float)
+    memb_size = np.array(memb_seg.shape).astype(float)
+    scale_nuc = raw_size / nuc_size
+    scale_memb = raw_size / memb_size
+
+    # Actual resize
+    nuc_seg = proc.resize(nuc_seg, scale_nuc, method="nearest")
+    memb_seg = proc.resize(memb_seg, scale_memb, method="nearest")
+
+    # Normalize images
+    normalized_images = []
+    for i, index in enumerate(channel_indices):
+        if index == brightfield_channel_index:
+            norm_method = "trans"
+        else:
+            norm_method = "img_bg_sub"
+
+        # Normalize and append
+        normalized_images.append(proc.normalize_img(raw[i], method=norm_method))
+
+    # Stack all together
+    img = np.stack([nuc_seg, memb_seg, *normalized_images])
+    channel_names = Channels.DefaultOrderList
+
+    return img, channel_names, tuple(desired_pixel_sizes)
+
+
+def select_and_adjust_segmentation_ceiling(
+    image: np.ndarray, cell_index: int, cell_ceiling_adjustment: int = 7
+) -> np.ndarray:
+    """
+    Select and adjust the cell shape "ceiling" for a specific cell in the provided
+    image.
+
+    Parameters
+    ----------
+    image: np.ndarray
+        The 4D, CYXZ, image numpy ndarray output from `get_normed_image_array`.
+
+    cell_index: int
+        The integer index for the target cell.
+
+    cell_ceiling_adjustment: int
+        The adjust to use for raising the cell shape ceiling. If <= 0, this will be
+        ignored and cell data will be selected but not adjusted.
+        Default: 7
+
+    Returns
+    -------
+    adjusted: np.ndarray
+        The image with the membrane segmentation adjusted for ceiling shape correction.
+
+    Notes
+    -----
+    The original version of this function can be found at:
+    https://aicsbitbucket.corp.alleninstitute.org/projects/MODEL/repos/image_processing_pipeline/browse/aics_single_cell_pipeline/utils.py#83
+    """
+    # Select only the data in the first two channels (the segmentation channels)
+    # where the data matches the provided cell index
+    image[0:2] = image[0:2] == cell_index
+
+    # Because they are conservatively segmented,
```

Comment:
To my understanding this has always been a problem with the membrane segmentation on integrated cell pipeline data. The membrane segmentation doesn't go all the way to the ceiling of the cell. @jxchen01 would have more insight here.

Reply:
this is the so-called "roof-augmentation" that Greg invented to handle the bad "roof" in old membrane segmentation. Specially, because the photobleaching, the signal near the top is very weak. Then, the membrane segmentation stops earlier (in terms of z position) than the truth. For some structures living near the top of the cell, like mitochodira, the structure segmentation may be out of the membrane segmentation, as membrane seg is "shorter" than it should be. Then, the structure segmetnation will be mostly choped off and make the integrated cell model learning nothing. So, "roof-augmentation" is the method to fix the "shorter" membrane segmentation issue. 

For the purpose of reproduing the results, we should keep this. But, after the new cell segmentation is uploaded to labkey. We may want to get rid of the "roof-augmentation".

Reply:
Thanks for that info. I will probably add that all as a comment in the code :)

Fortunately we won't need to change the code too much but rather just simply change the default `cell_ceiling_adjustment` parameter to 0.